zerver/lib/types.py

```diff
@@ -354,3 +357,28 @@ class AnalyticsDataUploadLevel(IntEnum):
     BASIC = 1
     BILLING = 2
     ALL = 3
+
+
+@dataclass
+class StreamMessageEditRequest:
+    is_content_edited: bool
+    is_topic_edited: bool
+    is_stream_edited: bool
+    is_message_moved: bool
+    topic_resolved: bool
+    topic_unresolved: bool
+    content: str
+    topic_name: str
+    pre_truncation_topic_name: str
+    stream: "Stream"
+    orig_content: str
+    orig_topic_name: str
+    orig_stream: "Stream"
+    propagate_mode: str
+
+
+@dataclass
+class DirectMessageEditRequest:
```

Comment:
Added two different types as direct messages do not have stream and topic related fields and we would need to keep those as optional for them which will require assert statements in the code to keep mypy happy.

Also, it was suggested to have different classes for content edit, topic edit and stream edit and then check the types, but since we allow changing multiple things in a single request, I went ahead with a single `StreamMessageEditRequest` class.

Reply:
I don't think we still allow changing content at the same time as anything else. So I think there are probably 6 types:
- Content edit for DM
- Content edit for channel message
- Resolve topic
- Change just topic for channel message
- Change just channel for channel message
- Change both topic and channel for channel message.

Reply:
API still allows allow topic and content together.